### PR TITLE
WIP: Make pirum generate a composer repository as well as a pear one

### PR DESCRIPTION
**Pull request by Seldaek**
This is not finished, but I'd like to get someone with more PEAR packages and more experience with Pirum than I do to try it already.

The main missing thing is the autoload, but it's a problem I still have to solve.

### CODE REVIEW

**Comment by fabpot** [comment]
This code looks good to me.

**Comment by ghost** [comment]
+1

**Comment by henrikbjorn** [comment]
+1 this is a nice idea

**Comment by Seldaek** [comment]
@fabpot: Finally had time to add the autoload stuff. I think it's mergeable now.

File: pirum [hunk 1]
```diff
@@ -535,6 +548,132 @@ pear install <?php echo $this->server->alias ?>/package_name-beta</code></pre>
...
+                    $version['require']['php'] = '>='.$release['php'];
+                }
+
+                if (isset($release['deps']) && $deps = @unserialize($release['deps'])) {
+                    if (isset($deps['required'])) {
+                        foreach ($deps['required'] as $dep => $spec) {
+                            if ('pearinstaller' == $dep) {
+                                continue;
+                            }
+                            $version['require'][$dep] = '>='.(isset($spec['min']) ? $spec['min'] : '0.0.0');
```
Comment (stof) [inline pirum:610]: @fabpot the way you build the requirement is wrong here. You should add the upper bound too (it generally does not make sense to release something allowing any future version as you don't know the future, so we should do it properly when `$spec['max']` is available)

**Comment by stof** [comment]
Btw, the current naming for composer packages is likely to create conflicts as it does not use the channel name but only the package name

**Comment by Seldaek** [comment]
It's handled in https://github.com/composer/composer/pull/520 as long as you add it as a pear repo. But anyway it won't conflict since no composer packages are just called `Foo`.

**Comment by stof** [comment]
@Seldaek It can conflict if someone imports the repo as a composer repo instead of a pear repo (meaning that the name won't be transformed) and uses several such repos (let think about `pear.php.net/Config` and `pear.symfony.com/Config`, even if getting the Sf2 components from the pirum-generated composer repo is a bit silly as they are on packagist)

**Comment by Seldaek** [comment]
Ok, sure, but the point of this PR is just to optimize PEAR repos in composer, nobody really has to know about this. Users should assume it's a PEAR repo, and then all is well, if it's a recent Pirum repo it will load faster in composer and that's all that changes.

**Comment by stof** [comment]
oh, so the goal was not letting them import it as a composer repo instead of a pear repo ?

**Comment by Seldaek** [comment]
No I don't think you should have to care, you just add a PEAR repo, if you have to check if it's a Pirum repo and then if it's a recent enough Pirum repo, it's just annoying.

**Comment by stof** [comment]
well, then OK about the naming. But the requirement constraints should be fixed as loading the packages.json file is not equivalent to loading the packages through the PEAR protocol in composer currently (as Composer handles them properly IIRC)

**Comment by Seldaek** [comment]
@stof see #46

**Pull request merged**